haystack/document_store/sql.py

```diff
@@ -440,6 +440,23 @@ def query_by_embedding(self,
                                   "Change the query type (e.g. by choosing a different retriever) "
                                   "or change the DocumentStore (e.g. to ElasticsearchDocumentStore)")
 
+    def delete_document_by_id(self, id: Union[List[str], str], index: Optional[str] = None):
```

Comment:
I think this function should be implemented into all the document stores like elastic, faiss, milvus etc. So we can make it abstract function in `BaseDocumentStore` class